impeller/typographer/glyph_atlas.h

```diff
@@ -213,20 +223,27 @@ class FontGlyphAtlas {
   /// @return     The location of the glyph in the atlas.
   ///             `std::nullopt` if the glyph is not in the atlas.
   ///
-  std::optional<std::pair<Rect, Rect>> FindGlyphBounds(
-      const SubpixelGlyph& glyph) const;
+  std::optional<FrameBounds> FindGlyphBounds(const SubpixelGlyph& glyph) const;
+
+  //----------------------------------------------------------------------------
+  /// @brief      Append the frame bounds of a glyph to this atlas.
+  ///
+  ///             This may indicate a placeholder glyph location to be replaced
+  ///             at a later time, as indicated by FrameBounds.placeholder.
+  void AppendGlyph(const SubpixelGlyph& glyph, const FrameBounds& frame_bounds);
 
  private:
   friend class GlyphAtlas;
+
   std::unordered_map<SubpixelGlyph,
-                     std::pair<Rect, Rect>,
+                     FrameBounds,
                      SubpixelGlyph::Hash,
                      SubpixelGlyph::Equal>
       positions_;
 
   FontGlyphAtlas(const FontGlyphAtlas&) = delete;
 
-  FontGlyphAtlas& operator=(const FontGlyphAtlas&) = delete;
+  // FontGlyphAtlas& operator=(const FontGlyphAtlas&) = delete;
```

Comment:
Did you mean to comment this out? Couldn't find where you don't want to copy but do want to make it copy assigned. At least the rule of 5 thing doesn't track.

Reply:
I forgot about this one. I wasn't sure if we should try to design around this. Basically with this I can't do:

```
  font_atlas_map_[scaled_font] = FontGlyphAtlas();
  return &font_atlas_map_[scaled_font];
```

and I'm not sure what the alternative is.

Reply:
I think the alternative is [`insert`](https://en.cppreference.com/w/cpp/container/unordered_map/insert). You provide a pair and it returns an iterator.